### ISSUE TITLE
Chromium for web service not working needs google chrome

### ISSUE THREAD

**Issue by minsis** [zabbix/zabbix-docker#891]
##### SUMMARY

Web Service container is installing chromium and according to your doc its not supported (or at least 'untested') by y'all. In my case with chromium installed it currently doesn't work to generate reports. Once I uninstall it and install google-chrome it works fine.

##### OS / ENVIRONMENT / Used docker-compose files

Using OL containers Zabbix version 5.4.6

**Comment by dotneft** [comment]
It works fine with chromium as well:
```
2021/11/04 16:02:58.390013 making chrome headless request with parameters url: http://zabbix-web-nginx-mysql:8080//zabbix.php?action=dashboard.print&dashboardid=1&from=2021-11-03%2000%3A00%3A00&to=2021-11-04%2000%3A00%3A00, width: 1920, height: 992 for report request from 172.16.239.3:38252
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:02:59 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/04 16:03:03.739648 writing response for report request from 172.16.239.3:38252
```

Did you specify capabilities?

**Comment by minsis** [comment]
Apologies, I thought I specified the issue I'm having in my OP.

When I have chromium installed I just receives a blank report. When I install google-chrome the report is fine.

**Comment by dotneft** [comment]
Got it! Will check :-)

**Comment by dotneft** [comment]
Just verified with 5.4.7 image (Oracle Linux): sha256:7f93dcdb23583abc9c87d9cf744d0f5b32ea985d7f38843b653595569e3ec465. Everything is fine :-)

How did you install Google Chrome?

**Comment by minsis** [comment]
Followed the instructions here: https://blog.zabbix.com/scheduled-report-generation-in-zabbix-5-4/14776.

I simply logged into the container and uninstalled chromium and then installed the google-chrome browser.

Maybe the 5.47 container has a better version of chromium I'll have to look.

**Comment by dotneft** [comment]
I checked 5.4.6 version. Did you provide capabilities (SYS_ADMIN)?

**Comment by minsis** [comment]
This is what I'm spinning up:

```yaml
zabbix-web-service:
    image: zabbix/zabbix-web-service:ol-5.4.6
    container_name: zabbix-web-service
    volumes:
      - ./instance/var/lib/zabbix/enc:/var/lib/zabbix/enc:ro
    cap_add:
      - SYS_ADMIN
    deploy:
      resources:
        limits:
          cpus: '0.5'
          memory: 512M
        reservations:
          memory: 256M
    env_file:
      - ./.zabbix-web-service.env
    networks:
      zbx_net_outbound:
      zbx_net_backend:
        aliases:
          - zabbix-web-service
          - zabbix-web-service-ol
    stop_grace_period: 5s
    labels:
      com.zabbix.description: "Zabbix web service"
      com.zabbix.company: "Zabbix LLC"
      com.zabbix.component: "web-service"
      com.zabbix.os: "oracle linux"

```

**Comment by dotneft** [comment]
what host os do you use?

**Comment by minsis** [comment]
Ol7

**Comment by dotneft** [comment]
Tried that, everything is fine :-) Do you have some custom things on OL7?

**Comment by minsis** [comment]
> Tried that, everything is fine :-) Do you have some custom things on OL7?

Maybe security configurations. Not sure what would make the difference between chromium and google-chrome that would cause issues. The only thing I can maybe think of is that google-chrome required some library dependencies but I would maybe assume that chromium has similar requirements. I'll try to poke around I guess.

In any case why is chromium being used in the official image when Zabbix states that its 100% untested? Unless that has changed or something. At this point I would be proof that chromium isn't as reliable as the google-chrome browser.

**Comment by minsis** [comment]
I upgraded my dev env (which is all containers running on one server) to 5.4.7 and it seems to work there.

I'm going to upgrade our production env (spread out) next week and see what happens there.

**Comment by dotneft** [comment]
Chromium - opensource, Chrome is not :-) It has additional components, which are not opensource: https://www.google.com/intl/en/chrome/terms/

**Comment by minsis** [comment]
I've upgraded our prod system and still have the same issue of blank reports. Not sure what I'm missing here compared to our test environment.

On the test environment I did install chromium on the OS itself but then later uninstalled it and its dependencies so I'm not sure if this is the issue.

I have a feeling that the base OS is missing some dependencies since if I install google-chrome it will also install some libraries. Just not sure which one(s).

**Comment by minsis** [comment]
Ok so continuing to mess around I switched to a different dashboard and it works. So I guess there's too much data happening on my dashboard and its timing out or something I dont know. With my larger dashboard the process only runs for 7 seconds before stopping and then errors out at the 11s mark.

```
2021/11/10 20:26:19.057362 received report request from 192.168.200.3:22755
2021/11/10 20:26:19.058195 making chrome headless request with parameters url: http://zabbix-web:8080/zabbix.php?action=dashboard.print&dashboardid=2&from=2021-11-09%2000%3A00%3A00&to=2021-11-10%2000%3A00%3A00, width: 1920, height: 4422 for report request from 192.168.200.3:22755
2021/11/10 20:26:20 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
...
2021/11/10 20:26:27 ERROR: could not unmarshal event: unknown PrivateNetworkRequestPolicy value
2021/11/10 20:26:31.048752 Cannot fetch data: context deadline exceeded.
```

**Comment by dotneft** [comment]
Please try to increase timeouts on both sides.

**Comment by minsis** [comment]
Increasing web-service timeout to 30s helped generate the larger dashboard. Thanks for the tip there.

But back to the original issue. Not sure what you want to do with the ticket as it seems issues were probably fixed in 5.4.7 - at least for me.

**Comment by dotneft** [comment]
It is strange, because 5.4.6 and 5.4.7 have exactly the same Dockerfile :-) But anyway, I'm happy that the issue is solved. I'm closing the ticket.
